### PR TITLE
WIP: BL-7659 Load HGV/PSV subscriptions for notification

### PR DESCRIPTION
**Pull request by michalf-kainos**

### CODE REVIEW

File: motr-subscription-loader/src/main/java/uk/gov/dvsa/motr/subscriptionloader/processing/producer/SubscriptionCriteria.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public LocalDate getTestDueDate() {
+        return testDueDate;
+    }
+
+    public VehicleType getVehicleType() {
+        return vehicleType;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (przemos) [inline motr-subscription-loader/src/main/java/uk/gov/dvsa/motr/subscriptionloader/processing/producer/SubscriptionCriteria.java]: maybe EqualsBuilder from ApacheCommons ?

File: motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/dynamodb/fixture/model/SubscriptionItem.java [hunk 2]
```diff
@@ -26,6 +27,8 @@
 
     private String dvlaId = RandomDataUtil.motTestNumber();
 
+    private VehicleType vehicleType = VehicleType.MOT;
```
Comment (przemos) [inline motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/dynamodb/fixture/model/SubscriptionItem.java]: would there be still a case where we need a default value here ?

File: motr-subscription-loader/src/main/java/uk/gov/dvsa/motr/subscriptionloader/processing/producer/SubscriptionCriteria.java (hunk 1 above)
Comment (michalf-kainos) [reply motr-subscription-loader/src/main/java/uk/gov/dvsa/motr/subscriptionloader/processing/producer/SubscriptionCriteria.java]: I went with the pattern already usend in [ContanctDetail](https://github.com/dvsa/motr/blob/c519c0569c042bf9322006272d21420ce58b189e/motr-notifier/src/main/java/uk/gov/dvsa/motr/notifier/processing/model/ContactDetail.java#L34|ContactDetail)

File: motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/dynamodb/fixture/model/SubscriptionItem.java (hunk 2 above)
Comment (michalf-kainos) [reply motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/dynamodb/fixture/model/SubscriptionItem.java]: Notifier tests require proper subscription loading, either the vehicleType would be set to default here or in every integration test in this module

Comment (przemos) [reply motr-notifier/src/integration-test/java/uk/gov/dvsa/motr/test/integration/dynamodb/fixture/model/SubscriptionItem.java]: ok

File: motr-subscription-loader/src/integration-test/java/uk/gov/dvsa/motr/test/integration/subscriptionloader/SubscriptionLoaderDvlaVehicleTests.java [hunk 3]
```diff
@@ -1,49 +1,37 @@
...
     @Before
     public void setUp() {
-
-        queueHelper = new SqsHelper();
-        eventHandler = new EventHandler();
-        subscriptionItem = new SubscriptionItem();
-        fixture = new DynamoDbFixture(dynamoDbClient());
+        super.setUp();
+        subscriptionItem = new SubscriptionItem()
+            .setVehicleType(VehicleType.MOT);
```
Comment (przemos) [inline motr-subscription-loader/src/integration-test/java/uk/gov/dvsa/motr/test/integration/subscriptionloader/SubscriptionLoaderDvlaVehicleTests.java]: no need to set VehicleType if it is MOT by default (see comment at the top)

File: motr-subscription-loader/src/integration-test/java/uk/gov/dvsa/motr/test/integration/subscriptionloader/SubscriptionLoaderTests.java [hunk 4]
```diff
@@ -1,111 +1,83 @@
...
     public void setUp() {
-
-        queueHelper = new SqsHelper();
-        eventHandler = new EventHandler();
-        subscriptionItem = new SubscriptionItem().generateMotTestNumber();
-        fixture = new DynamoDbFixture(dynamoDbClient());
+        super.setUp();
+        subscriptionItem = new SubscriptionItem()
+                .generateMotTestNumber()
+                .setVehicleType(VehicleType.MOT);
```
Comment (przemos) [inline motr-subscription-loader/src/integration-test/java/uk/gov/dvsa/motr/test/integration/subscriptionloader/SubscriptionLoaderTests.java]: again, no need to set it up explicitly if it is done by default

**Pull request merged**